src/cacheSeed/cacheSeedManager.ts

```diff
@@ -32,63 +36,95 @@ export class CacheSeedManager {
-        currentTaskAttemps: attempts,
-        jobId,
-        taskId,
-      });
-      await this.queueClient.queueHandlerForTileSeedingTasks.reject(jobId, taskId, false);
-      await this.queueClient.queueHandlerForTileSeedingTasks.jobManagerClient.updateJob(jobId, { status: OperationStatus.FAILED });
-      return false;
-    }
-    try {
-      await this.delay(this.gracefulReloadMaxSeconds);
-      await this.runTask(seeds, jobId, taskId);
-      await this.queueClient.queueHandlerForTileSeedingTasks.ack(jobId, taskId);
-      await this.queueClient.queueHandlerForTileSeedingTasks.jobManagerClient.updateJob(jobId, {
-        status: OperationStatus.COMPLETED,
-        percentage: 100,
-      });
-    } catch (error) {
-      await this.queueClient.queueHandlerForTileSeedingTasks.reject(jobId, taskId, true, (error as Error).message);
-    }
+        this.logger.info({
+          msg: `Found new seed job: ${jobId}`,
+          jobId,
+          taskId,
+          productId: job.resourceId,
+          productVersion: job.version,
+          productType: job.productType,
+        });
 
-    // complete the current pool
-    return Boolean(tilesTask);
-  }
+        if (attempts > this.seedAttempts) {
+          this.logger.warn({
+            msg: `Reached to max attempts and will close task as failed`,
+            maxServiceAttempts: this.seedAttempts,
+            currentTaskAttemps: attempts,
+            jobId,
+            taskId,
+          });
+          await this.queueClient.queueHandlerForTileSeedingTasks.reject(jobId, taskId, false);
+          await this.queueClient.queueHandlerForTileSeedingTasks.jobManagerClient.updateJob(jobId, { status: OperationStatus.FAILED });
+          return false;
+        }
+
+        try {
+          await this.delay(this.gracefulReloadMaxSeconds);
+          await this.runTask(seeds, jobId, taskId);
+          await this.queueClient.queueHandlerForTileSeedingTasks.ack(jobId, taskId);
+          await this.queueClient.queueHandlerForTileSeedingTasks.jobManagerClient.updateJob(jobId, {
+            status: OperationStatus.COMPLETED,
+            percentage: 100,
+          });
+        } catch (error) {
+          await this.queueClient.queueHandlerForTileSeedingTasks.reject(jobId, taskId, true, (error as Error).message);
+        }
 
+        // complete the current pool
+        return Boolean(tilesTask);
+      },
+      this.tracer,
+      'handleCacheSeedTask',
+      spanOptions
+    );
+  }
   public async delay(seconds: number): Promise<void> {
     this.logger.info(`waiting before executing by gracefulReloadRandomSeconds delay for -${seconds}- seconds `);
     await new Promise((resolve) => setTimeout(resolve, seconds * this.msToSeconds));
   }
 
+  // eslint-disable-next-line @typescript-eslint/member-ordering
+  @withSpanAsyncV4
   private async runTask(seedTasks: ISeed[], jobId: string, taskId: string): Promise<void> {
+    const spanActive = trace.getActiveSpan();
```

Comment:
can  you create a reusable method of get span options with parameters?

Reply:
too complicated, with loadash manipulation and will hide some explicit content you want to provide the other developer